docs/guides/implementation-guides/how-to-implement-dependency-injection.md

```diff
@@ -3,13 +3,149 @@ id: how-to-implement-dependency-injection
 title: How To Implement Dependency Injection
 ---
 
-import MvvmArchitectureDiagram from '/img/guides/implementation-guides/mvvm-architecture.png';
+# How To Implement Dependency Injection in Avalonia
 
-# How To Implement Dependency Injection
-
-Content in preparation.
+[Dependency injection (DI)](https://en.wikipedia.org/wiki/Dependency_injection) allows developers to write cleaner, more modular, and testable code. It accomplishes this by creating discrete services that are passed around/created as needed. It is especially useful when you are developing multi-platform applications that need specific service implementations for each platform.
 
 This guide will show you how to use Dependency Injection (DI) with _Avalonia UI_ and the MVVM pattern. 
 
-<img src={MvvmArchitectureDiagram} alt=""/>
+Let's assume that your MainViewModel has a dependency on IMyService. The constructor for ViewModel should looks like this:
+
+```csharp
+private readonly IMyService _myService;
+
+public MainViewModel(IMyService myService)
+{
+    _myService = myService;
+}
+```
+
+Typically you would directly instantiate `MyService` and pass it into `MainViewModel`, like this:
```

Comment:
You may need to do a very minimal implementation of the service to show `MyService` and prevent people complaining about the example not compiling. Otherwise you will need to reword everything else to refer to the interface.